xds/src/main/java/io/grpc/xds/GcpAuthenticationFilter.java

```diff
@@ -254,23 +259,38 @@ public void sendMessage(ReqT message) {}
 
   private static final class LruCache<K, V> {
 
-    private final Map<K, V> cache;
+    private Map<K, V> cache;
+    private int maxSize;
 
     LruCache(int maxSize) {
-      this.cache = new LinkedHashMap<K, V>(
-          maxSize,
-          0.75f,
-          true) {
-        @Override
-        protected boolean removeEldestEntry(Map.Entry<K, V> eldest) {
-          return size() > maxSize;
-        }
-      };
+      this.maxSize = maxSize;
+      this.cache = createEvictingMap(maxSize);
     }
 
     V getOrInsert(K key, Function<K, V> create) {
       return cache.computeIfAbsent(key, create);
     }
+
+    private void resizeCache(int newSize) {
+      if (newSize >= maxSize) {
+        maxSize = newSize;
+        return;
+      }
+      LinkedHashMap<K, V> newCache = createEvictingMap(newSize);
+      newCache.putAll(cache);
+      cache = newCache;
+      maxSize = newSize;
+    }
+
+    @SuppressWarnings("NonApiType")
```

Comment:
Why suppress the warning? Why not change the return value to Map? The `cache` field is already a Map.